server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java

```diff
@@ -73,23 +88,216 @@ public LoggerContextListener enableJulChangePropagation(LoggerContext loggerCont
+        return this;
+      }
+
+      public Builder setFileNamePrefix(String fileNamePrefix) {
+        checkFileName(fileNamePrefix);
+        this.fileNamePrefix = fileNamePrefix;
+        return this;
+      }
+
+      private static void checkFileName(String fileName) {
+        if (requireNonNull(fileName, "fileNamePrefix can't be null").isEmpty()) {
+          throw new IllegalArgumentException("fileNamePrefix can't be empty");
+        }
+      }
+
+      private static void checkProcessName(String fileName) {
+        if (requireNonNull(fileName, "processName can't be null").isEmpty()) {
+          throw new IllegalArgumentException("processName can't be empty");
+        }
+      }
+
+      public RootLoggerConfig build() {
+        checkProcessName(this.processName);
+        checkFileName(this.fileNamePrefix);
+        return new RootLoggerConfig(this);
+      }
+    }
+  }
+
+  public String buildLogPattern(LogbackHelper.RootLoggerConfig config) {
+    return LOG_FORMAT
+        .replace(PROCESS_NAME_PLACEHOLDER, config.getProcessName())
+        .replace(THREAD_ID_PLACEHOLDER, config.getThreadIdFieldPattern());
+  }
+
+  /**
+   * Creates a new {@link ConsoleAppender} to {@code System.out} with the specified name and log pattern.
+   *
+   * @see #buildLogPattern(RootLoggerConfig)
+   */
+  public ConsoleAppender<ILoggingEvent> newConsoleAppender(Context loggerContext, String name, String logPattern) {
     PatternLayoutEncoder consoleEncoder = new PatternLayoutEncoder();
     consoleEncoder.setContext(loggerContext);
-    consoleEncoder.setPattern(pattern);
+    consoleEncoder.setPattern(logPattern);
     consoleEncoder.start();
-    ConsoleAppender consoleAppender = new ConsoleAppender();
+    ConsoleAppender<ILoggingEvent> consoleAppender = new ConsoleAppender<>();
     consoleAppender.setContext(loggerContext);
     consoleAppender.setEncoder(consoleEncoder);
     consoleAppender.setName(name);
     consoleAppender.setTarget("System.out");
-    for (Filter filter : filters) {
-      consoleAppender.addFilter(filter);
-    }
     consoleAppender.start();
     return consoleAppender;
   }
 
+  /**
+   * Make logback configuration for a process to push all its logs to a log file.
+   * <p>
+   * <ul>
+   * <li>the file's name will use the prefix defined in {@link RootLoggerConfig#getFileNamePrefix()}.</li>
+   * <li>the file will follow the rotation policy defined in property {@link #ROLLING_POLICY_PROPERTY} and
+   * the max number of files defined in property {@link #MAX_FILES_PROPERTY}</li>
+   * <li>the logs will follow the specified log pattern</li>
+   * </ul>
+   * </p>
+   *
+   * @see #buildLogPattern(RootLoggerConfig)
+   */
+  public FileAppender<ILoggingEvent> configureGlobalFileLog(Props props, RootLoggerConfig config, String logPattern) {
+    LoggerContext ctx = getRootContext();
+    Logger rootLogger = ctx.getLogger(ROOT_LOGGER_NAME);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make the "rootLogger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1312)